File: apps/merchant-api/PaymentService/src/PaymentService.Application/Dtos/Orders/OrderItemWithCompletedDetailsDto.cs
namespace PaymentService.Application.Dtos.Orders;

public class OrderItemWithCompletedDetailsDto
{
    public string ProductName { get; set; } = string.Empty;
    public string Brand { get; set; } = string.Empty;
    public string Store { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public double BasePrice { get; set; }
    public OrderItemVariantDto OrderItemVariant { get; set; } = new();
}

File: apps/merchant-web/src/schemes/order/OrderItemWithCompletedDetailsDto.ts
import {OrderItemVariantDto} from "@/schemes/order/OrderItemVariantDto";

export class OrderItemWithCompletedDetailsDto {
  productName: string;
  brand: string;
  imageUrl: string;
  basePrice: number;
  orderItemVariant: OrderItemVariantDto;

  constructor(
    productName: string,
    brand: string,
    imageUrl: string,
    basePrice: number,
    orderItemVariant: OrderItemVariantDto
  ) {
    this.productName = productName;
    this.brand = brand;
    this.imageUrl = imageUrl;
    this.basePrice = basePrice;
    this.orderItemVariant = orderItemVariant;
  }
}